### PR TITLE
automate updates to apro

### PR DESCRIPTION
**Pull request by aidanhahn**
## Description
The enclosed modifications consist of the following
- a script that creates the relevant rel/vX.Y branch in apro if need be, bumps the emissary version referenced within, and then pushes the update
- commands calling this script in release/promote-oss/dev-to-rc

The script was tested locally, and is being merged to the shared iterative development branch so that we can improve upon it as we continue working on the other bits that need to be altered for the release pipeline.

### CODE REVIEW

File: releng/01-release-rc-update-apro [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    exit -1
+fi
+
+if [ -z "$2" ]; then
+    echo "Second argument must be a GA tag"
+    exit -1
+fi
+
+if [ -z "$APRO_REPO_TARGET" ]; then
+    echo "APRO_REPO_TARGET not set. This should be set in the environment"
```
Comment (acookin) [inline releng/01-release-rc-update-apro:14]: exit non zero?

File: builder/builder.mk [hunk 2]
```diff
@@ -793,6 +793,8 @@ release/promote-oss/dev-to-rc:
 		$(MAKE) VERSION_OVERRIDE=$${veroverride} push-manifests  ; \
 		$(MAKE) VERSION_OVERRIDE=$${veroverride} publish-docs-yaml ; \
 		$(MAKE) clean-manifests ; \
+		gavers=$$(echo $(RELEASE_VERSION) | sed 's/-rc.*//'); \
```
Comment (acookin) [inline builder/builder.mk:796]: you can also use $(VERSIONS_YAML_VERSION), and it's actually more reliable for -ea releases

**Review by acookin — CHANGES_REQUESTED** [review]
smol comments but otherwise looks good

File: .github/workflows/promote-rc.yml [hunk 3]
```diff
@@ -26,11 +28,16 @@ jobs:
       - name: "make release/promote-oss/dev-to-rc"
         run: |
           make release/promote-oss/dev-to-rc
+      - name: Update emissary tag in apro
```
Comment (acookin) [inline .github/workflows/promote-rc.yml:31]: you can probably make this a different job in the workflow. that way if this step fails we still get RC info and crap in the slack message

Comment (acookin) [reply .github/workflows/promote-rc.yml:31]: also this is double true becuase at this point the tree is dirty so RELEASE_VERSION won't be something you like i don't think

**Review by acookin — CHANGES_REQUESTED** [review]
one minor thing about moving that step into it's own job in the workflow

File: .github/workflows/promote-rc.yml (hunk 3 above)
Comment (aidanhahn) [reply .github/workflows/promote-rc.yml]: Good point!

File: .github/workflows/promote-rc.yml [hunk 4]
```diff
@@ -47,3 +46,18 @@ jobs:
...
             ]
+
+  update-apro-with-rc:
+    runs-on: ubuntu-latest
+    needs: promote-to-rc
+    name: update-apro-with-rc
+    env:
+      GIT_API_KEY: ${{ secrets.GH_API_KEY }}
+      APRO_REPO_TARGET: ${{ secrets.APRO_REPO_TARGET }}
+    steps:
```
Comment (acookin) [inline .github/workflows/promote-rc.yml:57]: you need to make sure to do the git credentials step in here too (and checkout)

**Review by acookin — CHANGES_REQUESTED** [review]

File: .github/workflows/promote-rc.yml [hunk 5]
```diff
@@ -47,3 +46,24 @@ jobs:
...
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: "setup git config"
+        run: |
+          set -x
+          git config --global user.email "dev@datawire.io"
```
Comment (anodelman) [inline .github/workflows/promote-rc.yml:64]: Let's get this into a secret along with the user.name - be easier if we ever have to change to a different account.

**Review by kflynn — CHANGES_REQUESTED** [review]
Looks sane, I think! I agree with @anodelman's commentary about secrets, though.

**Review by anodelman — APPROVED** [review]

**Pull request merged**